### DISCUSSION TITLE
What are some of the use-cases for using socketify.py?

### DISCUSSION THREAD

**Discussion by robkorv** [cirospaciari/socketify.py#96] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**

Clicked on a link to this project from PyCoder's Weekly #561 – JANUARY 24, 2023. It looks really interesting. But I'm missing context as to why this project exists and some examples of how it should be deployed or used in an production environment.

Would you use this behind another webserver like nginx for example? Or is this also a replacement for the webserver itself?

**Comment by cirospaciari** [comment, marked as answer]
Hey there.

This project is a web framework and WebServer and wants to be an all-in-one package solution for web development.
You can start fresh with socketify or use socketify WSGI and/or ASGI servers to boost your current project performance.

You can replace gunicorn + uvicorn + httptools + uvloop with socketify.
You can replace fastapi, flask, django, falcon, or any other Web Framework or use our ASGI and WSGI server with these frameworks too!

Here are some recent benchmarks in TechEmPower:
https://www.techempower.com/benchmarks/#section=test&runid=8b5c71ba-c738-444e-80d1-581f7e650b68&test=plaintext&l=hra0hr-35r

Socketify itself does not need ASGI and WSGI servers, that's why is so fast, I implemented ASGI and WSGI servers to boost the performance of all Web Frameworks.

in the future, Socketify will also implement fetch like API, web socket client, database connectors like MongoDB, PostgreSQL and Redis.

here are our docs: 
https://docs.socketify.dev/

and here is the part of our CLI tool, ASGI, and WSGI usage:
https://docs.socketify.dev/cli.html


We want to make web development with Python faster and more powerful, our focus is Performance and we want to bring PyPy with first-class support to be able to surpass GoLang Fiber.

You can use NGINX + Cloudflare for example to optimize our static files, but you don't need it. Socketify is really fast, and if you are writing APIS/Services, you don't need NGINX or anything like this.

For context, you can just deploy and run using the socketify cli tool, the production performance is out of the box, no need for more tooling.

Here are a lot of examples:
https://github.com/cirospaciari/socketify.py/tree/main/examples

You don't need to use CLI Tools (most of the examples don't use CLI tools) but you can if you prefer, the performance will be the same.

You can use it with Docker, NGINX, or pure socketify, your production setup can be done as you wish and prefer. I will recommend 
NGINX and a CDN only if you are serving static files.
